ocaml/parsing/jane_syntax.ml

```diff
@@ -186,11 +194,13 @@ module Comprehensions = struct
     let raise expr err = raise (Error(expr.pexp_loc, err))
   end
 
+  (* Returns the expression node with the outermost Jane Syntax-related
+     attribute removed. *)
   let expand_comprehension_extension_expr expr =
-    match Expression.match_jane_syntax expr with
-    | Some (comprehensions :: names, expr)
+    match find_and_remove_jane_syntax_attribute expr.pexp_attributes with
```

Comment:
I don't think I love this function name -- maybe we could restore `Expression.match_jane_syntax` by implementing it in terms of `with_attributes`?

Reply:
Given this only has one callsite, I CR-someday'ed its name as discussed